### PR TITLE
Feature/shap analysis

### PR DESCRIPTION
**Pull request by gvalleg**
Shap analysis V1:
- Handle binary classification, multiclass classification and regression problems.
- Make barplots (feature importance with SHAP), summary plots, waterfall plots and partial dependence plots.

### CODE REVIEW

File: mango/shap_analysis/shap_analysis.py [hunk 1]
```diff
@@ -0,0 +1,670 @@
...
+            "XGBRegressor",
+            "LGBMClassifier",
+            "LGBMRegressor",
+            "CatBoostClassifier",
+            "CatBoostRegressor",
+            "RandomForestClassifier",
+            "RandomForestRegressor",
+            "ExtraTreesClassifier",
+            "ExtraTreesRegressor",
+        ]
```
Comment (ggsdc) [inline mango/shap_analysis/shap_analysis.py:71]: I'm torn about these lists of "valid" values. I usually place them on a const.py file and import them from there. You can see an example of this in PR #99

File: mango/tests/shap_module/test_shap_analysis.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
+import shutil
```
Comment (ggsdc) [inline mango/tests/shap_module/test_shap_analysis.py:1]: These tests are not getting run

File: requirements-dev.txt [hunk 3]
```diff
@@ -1,2 +1,5 @@
 -r requirements.txt
```
Comment (ggsdc) [inline requirements-dev.txt:1]: Should we change pyproject.toml?

**Review by ggsdc — CHANGES_REQUESTED** [review]

File: mango/shap_analysis/shap_analysis.py (hunk 1 above)
Comment (gvalleg) [reply mango/shap_analysis/shap_analysis.py]: I have changed this

**Review by ggsdc — APPROVED** [review]

**Pull request merged**